File: problems/9/solution.cpp
ListNode* Solution::swapPairs(ListNode* head) {
    if (head == NULL || head->next == NULL) {
		return head;
	}

    ListNode* prev = head;
    ListNode* curr = head->next;

    head = curr;
    while (true) {
        ListNode *next = curr->next;
        curr->next = prev; // Change next of current as previous node

        if (next == NULL || next->next == NULL) {
			prev->next = next;
			break;
        }

        prev->next = next->next;

        prev = next;
        curr = prev->next;
    }
	return head;
}

File: problems/9/solution.java
// ListNode type
// class ListNode {
//     int val;
//     ListNode next;
//     ListNode(int x) { val = x; next = null; }
// }
class Solution {
    public static ListNode swapPairs(ListNode head) {
        if (head == null || head.next == null) {
            return head;
        }

        ListNode prev = head;
        ListNode curr = head.next;

        head = curr;
        while (true) {
            ListNode next = curr.next;
            curr.next = prev; // Change next of current as previous node

            if (next == null || next.next == null) {
                prev.next = next;
                break;
            }

            prev.next = next.next;

            prev = next;
            curr = prev.next;
        }
        return head;

    }
}
